aioredis/lock.py

```diff
@@ -1,7 +1,6 @@
 import asyncio
-import threading
+import contextvars
```

Comment:
contextvars isn't supported on pypy, so if we adopt this implementation, then we should simply not support locks for pypy implementations until we can find a solution.
```suggestion
try:
    import contextvars
except ImportError:
    pass
```